### PR TITLE
chore(stage): cleanup stage id

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Motivation

Having both `StageId` & `StageKind` is cumbersome & redundant.

## Description

* Remove `StageId` wrapper around a static `str`.
* Remove all `StageId` consts.
* Rename `StageKind` to `StageId` and port old methods there
* Move `StageId` to the primitives crate
* Expose `get_stage_checkpoint` helper function

This is cleanup work in preparation for `PipelineProvider` (or `StageCheckpointProvider`) to unblock #2679

### CODE REVIEW

File: crates/primitives/src/stage/id.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            StageId::MerkleExecute => "MerkleExecute",
+            StageId::MerkleUnwind => "MerkleUnwind",
+            StageId::Execution => "Execution",
+            StageId::TransactionLookup => "TransactionLookup",
+            StageId::Finish => "Finish",
+            StageId::Other(s) => s,
+        }
+    }
+
+    /// Returns true if it's a downloading stage [HEADERS] or [BODIES
```
Comment (onbjerg) [inline crates/primitives/src/stage/id.rs:59]:
```suggestion
    /// Returns true if it's a downloading stage [HEADERS] or [BODIES]
```

File: crates/primitives/src/stage/id.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/// All known stages
```
Comment (onbjerg) [inline crates/primitives/src/stage/id.rs:1]:
```suggestion
/// Stage IDs for all known stages.
///
/// For custom stages, use [`StageId::Other`]
```

**Review by onbjerg — APPROVED** [review]
small nit

**Review by onbjerg — APPROVED** [review]
needs lint - new lint in cargo, so i would advice to update

File: crates/primitives/src/stage/id.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
+/// Stage IDs for all known stages.
+///
+/// For custom stages, use [`StageId::Other`]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
```
Comment (gakonst) [inline crates/primitives/src/stage/id.rs:5]: We should reuse this in reth stage etc, let's add strum impls so that clap can parse this from cli

**Pull request merged**